zulip_bots/zulip_bots/bots/incrementor/incrementor.py

```diff
@@ -1,31 +1,33 @@
 # See readme.md for instructions on running this code.
 
+from zulip_bots.lib import ExternalBotHandler
 
 class IncrementorHandler(object):
     META = {
         'name': 'Incrementor',
         'description': 'Example bot to test the update_message() function.',
     }
 
-    def usage(self):
+    def usage(self) -> str:
         return '''
         This is a boilerplate bot that makes use of the
         update_message function. For the first @-mention, it initially
         replies with one message containing a `1`. Every time the bot
         is @-mentioned, this number will be incremented in the same message.
         '''
 
-    def initialize(self, bot_handler):
+    def initialize(self, bot_handler: ExternalBotHandler) -> None:
         storage = bot_handler.storage
         if not storage.contains('number') or not storage.contains('message_id'):
-            storage.put('number', 0)
-            storage.put('message_id', None)
+            # `storage.put` only accepts `str`s.
+            storage.put('number', str(0))
+            storage.put('message_id', '')
 
-    def handle_message(self, message, bot_handler):
+    def handle_message(self, message: dict, bot_handler: ExternalBotHandler) -> None:
         storage = bot_handler.storage
-        num = storage.get('number')
-        storage.put('number', num + 1)
-        if storage.get('message_id') is None:
+        num = int(storage.get('number'))
+        storage.put('number', str(num + 1))
```

Comment:
Does this not work already? Or is it a type issue?

Reply:
As of now, `storage.get('number')` is annotated to return a `str` only, so trying to `+ 1` it causes an exception. Therefore, casting it to an `int` is needed.

I think using `# type: ignore` could also work, but I thought it was better to cast it than ignore the type entirely.

Reply:
@skunkmb Ugh, the problem here is StateHandler.get/put are mis-annotated.  I'll fix this now.  I think your change is fine, though, as it's a bit more explicit.

Reply:
fixed in https://github.com/zulip/python-zulip-api/commit/b4ff5230a9947aed4c3c2c88cf22f618233190ce